[frontend] MSW の導入 #8

takafumi-nagashima-unicast opened the pull request:
## 概要
* msw の導入
    * msw と cross-env のインストール
    * `bun run dev:mock` で msw 起動
* チャットのやり取りを再現する mock handler として `sendVoiceChatHandlers.ts` を導入
    * チャットを送信すると今が何時何分なのか回答する
    * ResponseStatus として “status: 200” を返す
    * 10%の確率でメッセージの内容が変わる
* README 更新
    * `bun run dev:mock` でモックを起動する旨を記載

## 動作確認
* msw の起動
    * `/frontend` に移動、`bun run dev: mock` で起動すること
* チャットのやり取りと api リクエスト
    * `/api/zundamon/voice-chat` にリクエストを送る
    * メッセージを入力後、時報が返ってくる
        * 内容："今は 〇時 〇分 なのだ 🫛"
        * Response Status: `status: 200`
    * 10% の確率で返答内容が変わる
        * 内容："ｺﾛｼﾃ…ｺﾛｼﾃ…"
        * Response Status: `status: 200`

takafumi-nagashima-unicast committed:
feat: msw の構築と実装 #8
* msw と cross-env のインストール
* `bun run dev:mock` で msw 起動

takafumi-nagashima-unicast committed:
feat: sendVoiceChatHandlers.ts の実装 #8

- チャットを送信すると今が何時何分なのか回答する
- ResponseStatus として “status: 200” を返す
- 10%の確率でメッセージの内容が変わる

takafumi-nagashima-unicast committed:
docs: README の更新 #8

- `bun run dev:mock` でモックを起動する旨を記載

daichihayasaka commented on apps/frontend/src/mocks/handlers/sendVoiceChatHandlers.ts:
```diff
@@ -0,0 +1,39 @@
+        const minute = date.getMinutes();
+
+        const num = Math.random() * 10;
+        console.log("num: ", num);
```
コパイロット君のいう通り、特に理由がなければ削除でお願いします！
デバッグ用の console.log は残しておくとどんどん増えていきがちなので。。。

daichihayasaka commented on apps/frontend/src/mocks/handlers/sendVoiceChatHandlers.ts:
```diff
@@ -0,0 +1,39 @@
+        const num = Math.random() * 10;
+        console.log("num: ", num);
+
+        if (Math.trunc(num) == 4) {
```
ここでコメントは不要かな！

daichihayasaka commented on apps/frontend/src/mocks/handlers/sendVoiceChatHandlers.ts:
```diff
@@ -0,0 +1,39 @@
+        const hours = date.getHours();
+        const minute = date.getMinutes();
+
+        const num = Math.random() * 10;
```
細かい話ですがせっかくなので。。。
num という名前だと何を目的とした変数かわからないので、改善お願いします！

今回の場合むずいですが。。。、考える価値はあるかと！
理想的には変数名だけで目的わかるようにしたいですが、どうしても困難な場合、コメントで補強もアリかと思います！

daichihayasaka commented on apps/frontend/src/mocks/handlers/sendVoiceChatHandlers.ts:
```diff
@@ -0,0 +1,39 @@
+        const num = Math.random() * 10;
+        console.log("num: ", num);
+
+        if (Math.trunc(num) == 4) {
```
コパイロット君のいう通り、特に理由がなければ、 === でお願いします！

daichihayasaka reviewed (changes requested):
レビュー遅くなってすみません！
いくつかコメントしたので、ご確認お願いします！

takafumi-nagashima-unicast committed:
refactor: copilot 君の指摘箇所修正 #8

- 変数名を `num` → `dice` に変更
    - サイコロ的な意味で `dice`
- if の条件の強化
    - 値と型を比較 (厳密等価演算子)
- console.log() の削除
    - dice に入る数値の確認用 log を削除
    - デバッグ用のため

takafumi-nagashima-unicast commented:
@daichihayasaka 

copilot 君の指摘箇所について、修正し push いたしました。
ご確認お願いいたします。

37c3ac5

daichihayasaka reviewed (approved):
修正ありがとうございます！LGTM！
